RELEASE.md

```diff
@@ -41,58 +41,68 @@ with `make upgrade-npm-deps`
-
-- Entries in the `CHANGELOG.md` are meant to be in this order:
-
-* `[FEATURE]`
-* `[ENHANCEMENT]`
-* `[BUGFIX]`
-* `[BREAKINGCHANGE]`
-
-As we have many libraries we publish, it's better if you also put a clear indication about what library is affected by
+- Generate `CHANGELOG.md` updates based on git history:
+  ```bash
+  make generate-changelog
+  ```
+- Review the generated `CHANGELOG.md` for valid output. Things to check include:
+  - Entries in the `CHANGELOG.md` are meant to be in this order:
+    * `[FEATURE]`
+    * `[ENHANCEMENT]`
+    * `[BUGFIX]`
+    * `[BREAKINGCHANGE]`
+  - Entries that map to a pull request should include a pull request number.
+  - As we have many libraries we publish, it's better if you also put a clear indication about what library is affected by
 these changes.
-
-- Update the different `package.json` with the corresponding version:
-
-```bash
-make bump-version
-```
+  - Consumers understand how to handle breaking changes either through the messaging in the changelog or through the linked pull requests.
+- Update the `package.json` files for all packages with the corresponding version:
+  ```bash
+  make bump-version
+  ```
+- Push the branch to github and create a pull request with the release branch as the base. This gives others the opportunity to chime in on the release,
+  in general, and on the addition to the changelog, in particular. 
+  - It's also helpful to drop a link to the release PR in [#perses-dev](https://matrix.to/#/#perses-dev:matrix.org) to get extra visibility.
+- Address any necessary feedback.
+- Once the pull request is approved, merge the it into the release branch.
 
 ### 2. Draft the new release
 
-Tag the new release via the following commands:
-
-```bash
-git checkout release/v<version_to_be_replaced>
-make tag
-git push origin v<version_to_be_replaced>
-```
+- Pull down the latest updates to the release branch on your local machine to ensure you have the updates from the previous step.
+- Tag the new release via the following commands:
+  ```bash
+  git checkout release/v<version_to_be_replaced>
+  make tag
+  git push origin v<version_to_be_replaced>
+  ```
 
 Signing a tag with a GPG key is appreciated, but in case you can't add a GPG key to your Github account using the
 following [procedure](https://docs.github.com/en/authentication/managing-commit-signature-verification), you can replace
 the `-s` flag by `-a` flag of the git tag command to only annotate the tag without signing.
 
-Once a tag is created, the release process through the GitHub Actions will be triggered for this tag.
+Once a tag is created, an automated release process for this tag is triggered via Github Actions. This automated process includes:
+- Publishing new versions of the UI libraries to npm.
+- Building new go binaries and docker images.
+- Creating a new Github release that uses the changelog as the release notes and provides tarballs with the latest go binaries & docker images.
```

Comment:
@Nexucis is this bit accurate? Anything notable I'm missing? Are the docker images/binaries published anywhere else?

Reply:
the docker images are only published on docker hub:  https://hub.docker.com/r/persesdev/perses

They are not pushed to github if it's what you mean.

Reply:
Gotcha! Looks like just the go binaries are in those tarballs. I knew I wasn't getting this quite right. Will reword this.